nrf_security/src/random_ext.c

```diff
@@ -7,18 +7,31 @@
 
 #include "common.h"
 #include "nrf_cc3xx_platform_ctr_drbg.h"
+#include "nrf_cc3xx_platform_hmac_drbg.h"
 #include "psa/crypto.h"
 #include "psa/crypto_platform.h"
 
  psa_status_t mbedtls_psa_external_get_random(
     mbedtls_psa_external_random_context_t *context,
     uint8_t *output, size_t output_size, size_t *output_length)
 {
-    int ret = nrf_cc3xx_platform_ctr_drbg_get(
+    int ret;
+
+#if defined(CONFIG_PSA_WANT_ALG_CTR_DRBG)
+    ret = nrf_cc3xx_platform_ctr_drbg_get(
+        NULL,
+        output,
+        output_size,
+        output_length);
+#elif defined(CONFIG_PSA_WANT_ALG_HMAC_DRBG)
+    ret = nrf_cc3xx_platform_hmac_drbg_get(
         NULL,
         output,
         output_size,
         output_length);
+#else
+    #error "Enable CONFIG_PSA_WANT_ALG_CTR_DRBG or CONFIG_PSA_WANT_ALG_HMAC_DRBG"
+#endif
```

Comment:
random_ext.c shouldn't need to know that nrf_cc3xx_platform_ctr_drbg_get is invoked when CONFIG_PSA_WANT_ALG_CTR_DRBG is enabled.

We should have a dedicated function that knows which get function to call based on PSA defines and this function should be invoked from here.

The same applies to the init functions.

Reply:
Discussed offline and we will leave this as is for now.